label_studio/users/views.py

```diff
@@ -149,3 +158,28 @@ def user_account(request):
         'users/user_account.html',
         {'settings': settings, 'user': user, 'user_profile_form': form, 'token': token},
     )
+
+
+class UserSoftDeleteView(generics.RetrieveDestroyAPIView):
+    permission_required = ViewClassPermission(
+        DELETE=all_permissions.organizations_change,
+    )
+    queryset = User.with_deleted.all()
+    serializer_class = UserSerializer
+    permission_classes = (IsAuthenticated, HasObjectPermission)
+
+    def get_object(self):
+        pk = self.kwargs[self.lookup_field]
+        user = self.queryset.filter(active_organization=self.request.user.active_organization).get(pk=pk)
```

Comment:
Probably worth a comment that this line effectively stops users from being able to delete users in other organizations (I take it we can rely on `request.user.active_organization` for this?)

Reply:
makes sense.  Yes I'm pretty certain thats the field we can rely on, but we can check with @bmartel  as well to be safe.

Reply:
to this comment: https://github.com/HumanSignal/label-studio/pull/4876/files#r1353458664, I think the 500 is likely coming from this `get` call. One option would be to use `fast_first` on this queryset instead and handle the None case by raising an exception.